src/Microsoft.AspNetCore.Sockets/Internal/ConnectionState.cs

```diff
@@ -75,6 +80,9 @@ public ConnectionState(Connection connection, IChannelConnection<Message> applic
 
             // REVIEW: Add a timeout so we don't wait forever
             await Task.WhenAll(applicationTask, transportTask);
```

Comment:
No, I meant move this line to right after 69, then anyone calling `DisposeAsync` a second or more times would be on `await Lock.WaitAsync()` and when the `applicationTask` and `transportTask` finish the lock gets released and the other `DisposeAsync` callers finish